### PR TITLE
Update pyk to point to its own repository

### PR DESCRIPTION
**Pull request by goodlyrottenapple**
This should get nix working with the moved pyk repo

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -17,9 +17,16 @@
     ethereum-legacytests.url =
       "github:ethereum/legacytests/d7abc42a7b352a7b44b1f66b58aca54e4af6a9d7";
     ethereum-legacytests.flake = false;
+    pyk.url =
+      "github:runtimeverification/pyk/3cef4165e26b1662241c1b8d1d1edd5551413856";
```
Comment (goodlyrottenapple) [inline flake.nix:21]: note that this is currently hardcoded to the commit of this PR:
https://github.com/runtimeverification/pyk/pull/1

The pin to this specific commit should be removed once the above PR is merged + we will need to set up a CI mechanism to track and update the pyk dependency the same way we keep `deps/` submodules in sync

Comment (ehildenb) [reply flake.nix:21]: We can do this with a github workflow, see this example where ksummarize keeps a file in sync: https://github.com/runtimeverification/ksummarize/blob/master/.github/workflows/update-version.yml

KSummarize uses a template, but you could probably do a sed find/replace, or just a bash script that loops over every line in this file and finds the one with comment `REPLACE THIS LINE`, and emits the updated line with the version.

Comment (tothtamas28) [reply flake.nix:21]: Do tags work here? It would be more convenient to point to a tag rather than a commit hash.

**Review by tothtamas28 — APPROVED** [review]

File: flake.nix (hunk 1 above)
Comment (goodlyrottenapple) [reply flake.nix:21]: I think so. Even if not, it should not be too difficult to make a script that queries git for the hash of a particular tag and then feeds that to nix.

**Pull request merged**